src/partitioning/bvh.rs

```diff
@@ -135,12 +138,15 @@ pub trait BVH<T, BV> {
                     match visitor.visit(best_cost, child_bv, child_data) {
                         BestFirstVisitStatus::Continue { cost, result } => {
                             if cost < best_cost {
+                                // result is_some iff child is a leaf node
```

Comment:
This is not correct. In the general case, we could accept child that contain a result but that is not a leaf.
```suggestion
```